robot-server/robot_server/protocols/router.py

```diff
@@ -75,6 +75,9 @@ class ProtocolFilesInvalid(ErrorDetails):
 )
 async def create_protocol(
     files: List[UploadFile] = File(...),
+    # optional protocol_key to track protocols by the UI.
+    # needs to be Form type because its formData
+    protocol_key: Optional[str] = Form(None),
```

Comment:
Let's just call the filed `key`. The client is already hitting `POST /protocols`, they know the key is for a protocol

```suggestion
    # use Form because request is multipart/form-data
    # https://fastapi.tiangolo.com/tutorial/request-forms-and-files/
    key: Optional[str] = Form(None),
```